doc/articles/Surfaces.md

```diff
@@ -0,0 +1,256 @@
+ - `<canvas>` HTML elements in Emscripten (targeting WebAssembly) with `WGPUSurfaceSourceCanvasHTMLSelector_Emscripten`
+ - `HWND` on Windows with @ref WGPUSurfaceSourceWindowsHWND
+ - `Window` using Xlib with @ref WGPUSurfaceSourceXlibWindow
+ - `wl_surface` on Wayland systems with @ref WGPUSurfaceSourceWaylandSurface
+ - `xcb_window_t` using XCB windows with @ref WGPUSurfaceSourceXCBWindow
+
+For example, creating an @ref WGPUSurface from an `HWND` is done like so:
+
+```c
+WGPUSurfaceSourceWindowsHWND hwndDesc = {
+    .chain = { .sType = WGPUSType_SurfaceSourceWindowsHWND, },
+    .hinstance = GetModuleHandle(nullptr),
+    .hwnd = myHWND,
+};
+WGPUSurfaceDescriptor desc {
+    .nextInChain = &hwndDesc.chain,
+    .label = "Main window",
+};
+WGPUSurface surface = wgpuInstanceCreateSurface(myInstance, &desc);
+```
+
+In addition, @ref WGPUSurfaces have a bunch of internal fields that could be represented like this (in C/Rust-like pseudocode):
+
+```cpp
+struct WGPUSurface {
+    // The parent object
+    WGPUInstance instance;
+
+    // The current configuration
+    Option<WGPUSurfaceConfiguration> config = None;
+
+    // The frame, if any.
+    Option<WGPUTexture> currentFrame = None;
+};
+```
+
+The behavior of `::wgpuInstanceCreateSurface``(instance, descriptor)` is:
+
+ - If any of these validation steps fails, return an error @ref WGPUSurface:
+
+    - Validate that all the sub-descriptors in the chain for `descriptor` are known to this implementation.
+    - Validate that `descriptor` contains information about exactly one OS surface.
+    - As best as possible, validate that the OS surface described in the descriptor is valid (for example a zero `HWND` doesn't exist and isn't valid).
+
+ - Return a new @ref WGPUSurface with its `instance` member initialized with the `instance` parameter and other values defaulted.
+
+## Querying Surface Capabilities {#Surface-Capabilities}
+
+Depending on the OS, GPU used, backing API for WebGPU and other factors, different capabilities are available to render and present the @ref WGPUSurface.
+For this reason, negotiation is done between the WebGPU implementation and the application to choose how to use the @ref WGPUSurface.
+This first step of the negotiation is querying what capabilities are available using `::wgpuSurfaceGetCapabilities` that fills an @ref WGPUSurfaceCapabilities structure with the following information:
+
+ - A bit set of supported @ref WGPUTextureUsage that are guaranteed to contains @ref WGPUTextureUsage_RenderAttachment.
+ - A list of supported @ref WGPUTextureFormat values, in order of preference.
+ - A list of supported @ref WGPUPresentMode values (guaranteed to contain @ref WGPUPresentMode_Fifo).
+ - A list of supported @ref WGPUCompositeAlphaMode values (@ref WGPUCompositeAlphaMode_Auto is always supported but never listed in capabilities as it just lets the implementation decide what to use).
```

Comment:
Are present modes and alpha modes are returned in order of preference as well?

Reply:
We didn't add constraints on their order because they have a clear default value. For formats we wanted to let people know that using formats[0] for when they don't know is a good choice.